### DISCUSSION TITLE
How to read the output ImagemeMagick stream in VBScript from WScript.Shell ?

### DISCUSSION THREAD

**Discussion by AlexLaforge** [ImageMagick/ImageMagick#7081] [Help]
### ImageMagick version

7.1.0-19

### Operating system, version and so on

Windows 10 or 2019 + VBScript/Classic ASP

### Description

ImageMagick for Windows does not provides a COM+ interface anymore since version 7.1.0-19.

Previously, using it in VBScript was easy, and permitted to convert the output image to a file, or to **get it directly from memory**, by assigning it to a "Classic ASP 1D Array acting like an in-memory blob" (instead of a file), like that :

```
'VBScript 1D Array acting like an in-memory blob
'ImageMagick result of the "Convert()" method will be stored in this variable.
Dim a_outputImgData
a_outputImgData = Empty

'ImageMagick component instance
Dim IM
Set IM = Server.CreateObject("ImageMagickObject.MagickImage.1")

    IM.Convert _
        "-size", "100x100", _
        "xc:", "-channel", "G", "+noise", "Random", _
        "-format", "jpg", "-strip", "-density", "72x72", "-quality", "60%", _
    a_outputImgData

Set IM = Nothing
IM = Empty

'Here in Classic ASP : Send the output image from the "VBScript 1D Array acting like an in-memory blob" to the Browser
Response.ContentType = "image/jpeg"
Response.BinaryWrite a_outputImgData
```

Now, we are stuck with the usage of WScript.Shell, like this sample coming from the [ImageMagick download page](https://imagemagick.org/script/download.php) :

```
Set objShell = wscript.createobject("wscript.shell")
objShell.Exec("magick ""e:/myimages/image.png"" ""e:/myimages/image.jpg""")
```

It is not suitable and efficient to write the file to the disk, then opening it again to finally stream it to the browser !

How can we "read/get/obtain/pipe" the output produced by the ImagemeMagick command-line (executed in WScript.Shell) directly from our parent VBScript or Classic ASP script ?

**Comment by dlemstra** [comment]
You can't, you will be going to need to use files.

**Reply by AlexLaforge** [reply]
What a pity. Providing interfaces for Lisp (for example) and not anymore for COM+ was just a bad decision.

**Reply by dlemstra** [reply]
We already explained to you why support for this was removed here: #4913.
